src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -479,6 +519,12 @@ private async Task EnqueueFullProjectWorkItemAsync(Project project, InvocationRe
 
                 foreach (var documentId in project.AnalyzerConfigDocumentIds)
                     await EnqueueDocumentWorkItemAsync(project, documentId, document: null, invocationReasons).ConfigureAwait(false);
+
+                if (WorkspaceConfigurationOptions.ShouldConnectSourceGeneratedFilesToWorkspace(project.Solution.Options))
```

Comment:
A comment here for the conditioning under the feature flag may be appropriate: IIRC you mentioned in private conversation this was mostly to limit risk and although technically this behavior doesn't depend on that flag it was easy to just piggy back on it. That logic is entirely sound, but it'd be good to put that in words here.